src/OpenLoco/src/World/Station.cpp

```diff
@@ -801,12 +805,73 @@ namespace OpenLoco
         }
     }
 
+    // TODO: Deduplicate
+    static constexpr std::array<Gfx::Font, 4> kZoomToStationFonts = {
+        Gfx::Font::medium_bold,
+        Gfx::Font::medium_bold,
+        Gfx::Font::small,
+        Gfx::Font::small,
+    };
+
+    static constexpr std::array<int16_t, 4> kZoomToBorder = {
+        6,
+        6,
+        2,
+        2,
+    };
+
+    static constexpr std::array<int16_t, 4> kZoomToTextHeight = {
+        11,
+        11,
+        7,
+        7,
+    };
+
     // 0x0048DCA5
     void Station::updateLabel()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
-        call(0x0048DCA5, regs);
+        // duplicate of drawStationName in Viewport.cpp
+        char buffer[256]{};
+        FormatArguments args{};
+        args.push(town);
+        auto* strEnd = StringManager::formatString(buffer, 256, name, args);
+
+        *strEnd++ = ' ';
+        *strEnd = '\0';
+
+        const auto remainingLength = strEnd - buffer;
+        StringManager::formatString(strEnd, remainingLength, getTransportIconsFromStationFlags(flags));
+
+        auto& drawingCtx = Gfx::getDrawingEngine().getDrawingContext();
+        auto tr = Gfx::TextRenderer(drawingCtx);
+
+        for (auto zoom = 0U; zoom < 4; ++zoom)
```

Comment:
i woulda though we have `kZoomLevels` but seems like we don't :( would be nice to add it :)